src/Templates/MethodTemplate.cshtml

```diff
@@ -162,9 +161,7 @@ func (client @(Model.Owner)) @(Model.PreparerMethodName)(@(Model.MethodParameter
 </text>
 }
 
-    @(Model.IsLongRunningOperation()
-        ? "return preparer.Prepare(&http.Request{Cancel: cancel})"
-        : "return preparer.Prepare(&http.Request{})")
+    return preparer.Prepare((&http.Request{}).WithContext(ctx))
 }
```

Comment:
I would add a check for nil contexts

Reply:
withContext will Panic if the context is `nil`. Is there any point in having an extra check and presumably to return an error ?

Reply:
I just wonder if it is a good idea to set the context to `context.Background()` if the user is passing a nil context, just for it not to panic.